src/shared/tests.h

```diff
@@ -177,7 +177,8 @@ _noreturn_ void log_test_failed_internal(const char *file, int line, const char
         ({                                                                                                              \
                 typeof(expr) _result = (expr);                                                                          \
                 if (_result < 0)                                                                                        \
-                        log_test_failed("Expected \"%s\" to succeed, but got error: %s", #expr, STRERROR(_result));     \
+                        log_test_failed("Expected \"%s\" to succeed, but got error: %d/%s",                             \
+                                        #expr, _result, STRERROR(_result));                                             \
```

Comment:
Currently it seems only used for functions that returns int, but potentially, it can be others, e.g. ssize_t. Please use
```c
char _sexpr[DECIMAL_STR_MAX(typeof(expr))];
xsprintf(_sexpr, DECIMAL_STR_FMT(_result), _result);
```
like some other places.

Or, maybe enough to use %PRIiMAX and cast the result to intmax_t.